OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -358,7 +358,7 @@ void RefreshServerList()
 
 				lanGames = lanGames.GroupBy(gs => gs.Address).Select(g => g.Last()).ToList();
 
-				Game.RunAfterTick(() => RefreshServerListInner(games.Concat(lanGames).ToList()));
+				Game.RunAfterTick(() => RefreshServerListInner(games == null ? (lanGames.Count == 0 ? null : lanGames.ToList()) : games.Concat(lanGames).ToList()));
```

Comment:
This line is a monster.  You can slay the beast by initializing `games` as an empty list instead of null, and then changing this and the line above to:
```csharp
games.AddRange(lanGames.GroupBy(gs => gs.Address).Select(g => g.Last()));
Game.RunAfterTick(() => RefreshServerListInner(games));
```

Then adapt `RefreshServerListInner` to check for an empty list instead of null.  I think it would be best to push this all the way into `LoadGameRows` so that you can:
```csharp
ScrollItemWidget nextServerRow = null;
var rows = LoadGameRows(games, out nextServerRow);
```